src/spikeinterface/core/node_pipeline.py

```diff
@@ -504,19 +533,21 @@ def _compute_peak_pipeline_chunk(segment_index, start_frame, end_frame, worker_c
             # to handle compatibility peak detector is a special case
             # with specific margin
             #  TODO later when in master: change this later
-            extra_margin = max_margin - node.get_trace_margin()
-            if extra_margin:
-                trace_detection = traces_chunk[extra_margin:-extra_margin]
-            else:
-                trace_detection = traces_chunk
-            node_output = node.compute(trace_detection, start_frame, end_frame, segment_index, max_margin)
+            extra_margin_left = left_margin - node.get_trace_margin()
+            extra_margin_right = right_margin - node.get_trace_margin()
+            trace_detection = traces_chunk[extra_margin_left : num_samples_chunk - extra_margin_right]
+            node_output = node.compute(
+                trace_detection, start_frame, end_frame, segment_index, left_margin, right_margin
+            )
             # set sample index to local
-            node_output[0]["sample_index"] += extra_margin
+            node_output[0]["sample_index"] += extra_margin_left
         elif isinstance(node, PeakSource):
-            node_output = node.compute(traces_chunk, start_frame, end_frame, segment_index, max_margin)
+            node_output = node.compute(traces_chunk, start_frame, end_frame, segment_index, left_margin, right_margin)
         else:
             # TODO later when in master: change the signature of all nodes (or maybe not!)
-            node_output = node.compute(traces_chunk, *node_input_args)
+            node_output = node.compute(
+                traces_chunk, start_frame, end_frame, segment_index, left_margin, right_margin, *node_input_args
+            )
```

Comment:
camarade, I am not super with this change.
A main design of nodes and local compute was it was separate from the start/stop and segment index.
In short a bit like a computing kernel. Why do we need to had theses 5 new args for all nodes ?

Reply:
Not for all nodes, but spike amplitudes need at least the margins!